pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/defaultvalues/ATDefaultValues.java

```diff
@@ -82,54 +91,44 @@ private boolean isRegistrationComplete(Activities activity) {
         String hruid = esProfile.getHruid();
-        return insertParticipantData(Map.of(GENOME_STUDY_CPT_ID, PREFIX.concat(getGenomicIdValue(hruid))), ddpParticipantId,
-                GENOME_STUDY_FIELD_TYPE);
+        insertParticipantData(GENOME_STUDY_FIELD_TYPE,
+                Map.of(GENOME_STUDY_CPT_ID, GENOMIC_ID_PREFIX.concat(getGenomicIdValue(hruid))), ddpParticipantId);
     }
 
     private String getGenomicIdValue(String hruid) {
-        this.setDataAccess(new BookmarkDao());
-        Optional<BookmarkDto> maybeGenomicId = ((BookmarkDao) dataAccess).getBookmarkByInstance(AT_GENOMIC_ID);
+        BookmarkDao bookmarkDao = new BookmarkDao();
+        Optional<BookmarkDto> maybeGenomicId = bookmarkDao.getBookmarkByInstance(AT_GENOMIC_ID);
         return maybeGenomicId.map(bookmarkDto -> {
-            ((BookmarkDao) dataAccess).updateBookmarkValueByBookmarkId(bookmarkDto.getBookmarkId(), bookmarkDto.getValue() + 1);
+            bookmarkDao.updateBookmarkValueByBookmarkId(bookmarkDto.getBookmarkId(), bookmarkDto.getValue() + 1);
             return String.valueOf(bookmarkDto.getValue());
         }).orElse(hruid);
     }
 
-    private boolean insertExitStatusForParticipant() {
-        String ddpParticipantId = elasticSearchParticipantDto.getProfile().orElseThrow().getGuid();
-        String datstatExitReasonDefaultOption = getDefaultExitStatus();
-        return insertParticipantData(Map.of(EXIT_STATUS, datstatExitReasonDefaultOption), ddpParticipantId, AT_PARTICIPANT_EXIT);
+    private void insertExitStatusForParticipant(String ddpParticipantId) {
+        String defaultValue = getDefaultExitStatus();
+        if (StringUtils.isBlank(defaultValue)) {
+            throw new DsmInternalError("No default exit status found for AT study");
+        }
+        insertParticipantData(AT_PARTICIPANT_EXIT, Map.of(EXIT_STATUS, defaultValue), ddpParticipantId);
     }
 
     private String getDefaultExitStatus() {
-        this.setDataAccess(FieldSettingsDao.of());
+        FieldSettingsDao fieldSettingsDao = FieldSettingsDao.of();
         Optional<FieldSettingsDto> fieldSettingByColumnNameAndInstanceId =
-                ((FieldSettingsDao) dataAccess).getFieldSettingByColumnNameAndInstanceId(Integer.parseInt(instance.getDdpInstanceId()),
-                        EXIT_STATUS);
+                fieldSettingsDao.getFieldSettingByColumnNameAndInstanceId(instanceId, EXIT_STATUS);
         return fieldSettingByColumnNameAndInstanceId.map(fieldSettingsDto -> {
             FieldSettings fieldSettings = new FieldSettings();
             return fieldSettings.getDefaultValue(fieldSettingsDto.getPossibleValues());
         }).orElse(StringUtils.EMPTY);
     }
 
-    private boolean insertParticipantData(Map<String, String> data, String ddpParticipantId, String fieldTypeId) {
-        this.setDataAccess(new ParticipantDataDao());
+    private void insertParticipantData(String fieldTypeId, Map<String, String> data, String ddpParticipantId) {
```

Comment:
Nice, adding `fieldTypeId` means that clients don't need to fumble around getting the whole `data` object and then (re)setting keys.  They can just say "set field X to Y".